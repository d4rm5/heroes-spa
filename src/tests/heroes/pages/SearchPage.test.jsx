import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter, useNavigate } from 'react-router-dom';
import { SearchPage } from '../../../heroes/pages/SearchPage';

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate,
}));

describe('Pruebas en SearchPage.jsx', () => {
  test('should print default values', () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
  test('should print Batman and the input with query value', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/search?q=Batman']}>
        <SearchPage />
      </MemoryRouter>
    );
    const input = screen.getByRole('textbox');
    expect(input.value).toBe('Batman');
    const img = screen.getByRole('img');
    expect(img.src).toBe('http://localhost/heroes/dc-batman.jpg');
    const alertPrimary = screen.getByTestId('alertPrimary');
    expect(alertPrimary.style.display).toBe('none');
    const alertDanger = screen.getByTestId('alertDanger');
    expect(alertDanger.style.display).toBe('none');
  });
  test('should print error if dont found heroe', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/search?q=Batman123']}>
        <SearchPage />
      </MemoryRouter>
    );
    expect(screen.getByText('Not hero with')).toBeTruthy();
  });
  test('should call navigate to new screen', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/search?q=Batman123']}>
        <SearchPage />
      </MemoryRouter>
    );
    const input = screen.getByRole('textbox');
    fireEvent.input(input, { target: { value: 'Spider' } });
    fireEvent.submit(input);
    expect(mockedUseNavigate).toHaveBeenCalledWith('?q=spider');
  });
});
