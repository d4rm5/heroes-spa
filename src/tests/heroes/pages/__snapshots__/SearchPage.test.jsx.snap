// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en SearchPage.jsx should print default values 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary"
        data-testid="alertPrimary"
      >
        Search a hero
      </div>
      <div
        class="alert alert-danger"
        data-testid="alertDanger"
        style="display: none;"
      >
        Not hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
